test/integration/ws/common.dart

```diff
@@ -254,6 +254,18 @@ void runCommonWebSocketIntegrationTests(
     expect(doneEventReceived, isFalse);
   });
 
+  test('should not close if the only listener is canceled', () async {
+    var webSocket = await connect(echoUri);
+
+    var subscription = webSocket.listen((_) {});
+    await subscription.cancel();
+
+    // Should still be able to add events.
+    webSocket.add('one');
+    expect(webSocket.closeCode, isNull);
+    expect(webSocket.closeReason, isNull);
```

Comment:
Yeah there isn't a good way to make that assertion since I closed the subscription. Asserting that the websocket is still open (via these to expectations here) and implicitly that a `StateError` wasn't thrown should be sufficient.